stdlib/ostruct.rb

```diff
@@ -60,10 +66,52 @@ def to_n
   def hash
     @table.hash
   end
+  
+  def table
+    @table
+  end
+  
+  def delete_field(name)
+    sym = name.to_sym
+    begin
+      singleton_class.__send__(:remove_method, sym, "#{sym}=")
+    rescue NameError
+    end
+    @table.delete sym
+  end
+  
+  def new_ostruct_member(name)
+    name = name.to_sym
+    unless respond_to?(name)
+      define_singleton_method(name) { @table[name] }
+      define_singleton_method("#{name}=") { |x| @table[name] = x }
+    end
+    name
+  end
 
+  `var ostruct_ids;`
+  
   def inspect
-    "#<#{self.class}: #{each_pair.map {|name, value|
-      "#{name}=#{self[name].inspect}"
-    }.join(" ")}>"
+    top = `(ostruct_ids === undefined)`
+    ostruct_id = `self.$object_id()`
```

Comment:
This should be Ruby, never call methods with their mangled name since method definitions could change any time.

Also never use `object_id`, always use `__id__`, just make it `self.__id__`.